### PR TITLE
Glue42 integration

### PR DESCRIPTION
**Pull request by drodriguez-Tick42**
Glue42 integration

### CODE REVIEW

File: src/client/src/rt-components/platform/adapters/glue/window.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
+import { WindowConfig } from '../types'
+import { Glue42 as Glue42Interface, Glue42 } from 'tick42-glue'
```
Comment (my-code-is-lit) [inline src/client/src/rt-components/platform/adapters/glue/window.ts]: Add type declarations for `tick42-glue`

File: src/client/src/rt-components/platform/adapters/glue/window.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+import { frameButtonBase64 } from './utils/frameButtonImage'
+import { onGlueLoaded } from './glue'
+
+type BrowserWindowProps = WindowConfig
+type GDWindow = Glue42Interface.Windows.GDWindow
```
Comment (my-code-is-lit) [inline src/client/src/rt-components/platform/adapters/glue/window.ts]: Define shape of the Glue42Interface in module declaration

File: src/client/src/rt-components/platform/adapters/glue/glue.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import Glue, { Glue42 as Glue42Interface } from 'tick42-glue'
```
Comment (my-code-is-lit) [inline src/client/src/rt-components/platform/adapters/glue/glue.ts]: Add type declarations `tick42-glue.d.ts` for these imports

File: src/client/src/rt-components/platform/adapters/glue/glue.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import Glue4Office, { Glue42Office as Glue42OfficeInterfaceWrapper } from 'glue4office'
```
Comment (my-code-is-lit) [inline src/client/src/rt-components/platform/adapters/glue/glue.ts]: Add type declarations for `glue4office`

File: src/client/package.json [hunk 5]
```diff
@@ -158,7 +158,9 @@
     "webpack": "4.39.1"
   },
   "optionalDependencies": {
-    "openfin-cli": "^2.0.4"
+    "glue4office": "1.4.5",
```
Comment (my-code-is-lit) [inline src/client/package.json]: if possible, can these be `npm` packages? as opposed to having to manually symlink them?

Comment (drodriguez-Tick42) [reply src/client/package.json]: Yes, the glue.js package is published as `@glue42/desktop` and the Glue4Office package is planned to be published soon:

```
  "@glue42/desktop": "4.5.7",
  "glue4office": "file:./glue4office-1.4.8.tgz",
```

**Comment by my-code-is-lit** [comment]
New PR to be made on separate branch.

**Pull request closed without merging**